devicetypes/smartthings/ozom-smart-siren.src/ozom-smart-siren.groovy

```diff
@@ -127,32 +146,37 @@ def configure() {
 
 def both() {
 	log.debug "both()"
-	on()
+	state.alarmCmd = ALARM_BOTH
+	def warningDuration = state.maxDuration ? state.maxDuration : DEFAULT_MAX_DURATION
+	state.lastDuration = warningDuration
+
+	zigbee.command(IAS_WD_CLUSTER, COMMAND_IAS_WD_START_WARNING, MODE_BOTH, DataType.pack(warningDuration, DataType.UINT16), STROBE_DUTY_CYCLE, STROBE_LEVEL)
 }
 
 def siren() {
 	log.debug "siren()"
-	on()
-}
 
-def strobe() {
-	log.warn "strobe() is not supported by this device"
+	state.alarmCmd = ALARM_SIREN
+	def warningDuration = state.maxDuration ? state.maxDuration : DEFAULT_MAX_DURATION
+	state.lastDuration = warningDuration
+
+	// start warning, burglar mode, no strobe, siren very high
+	zigbee.command(IAS_WD_CLUSTER, COMMAND_IAS_WD_START_WARNING, MODE_SIREN, DataType.pack(warningDuration, DataType.UINT16), "00", "00")
 }
 
-def on() {
```

Comment:
please keep the `on` function, as the device needs to support all the switch capabilities as well

Reply:
I've added on() function, please check